src/main.js

```diff
@@ -525,6 +526,44 @@ class MiniGraphCard extends LitElement {
       </svg>`;
   }
 
+  renderGridLines() {
+    const {
+      height, hours_to_show, grid_line_type,
+    } = this.config;
+
+    const containerWidth = 500;
+    let numLines;
+    let xRatio;
+
+    switch (grid_line_type) {
+      case 'hour':
+      default:
+        numLines = hours_to_show;
+        xRatio = containerWidth / numLines;
+        break;
+      case 'day':
+        numLines = Math.round(hours_to_show / 24);
+        xRatio = (containerWidth / numLines);
+        break;
+      case 'week':
+        numLines = Math.round(hours_to_show / 168);
+        xRatio = containerWidth / numLines;
+        break;
+    }
+    const lines = [];
+
+    for (let i = 0; i < numLines; i += 1) {
+      const x = xRatio * (i + 0.5);
```

Comment:
Assume we have `hours_to_show=2` & `grid_line_type=hour`.
Then `numLines  = 2` & `xRatio = 250`.
Then we will have 2 lines - for `125` & `375`, i.e. for `hour=0.5` & `hour=1.5`.
And the 1st line will be "thin" (0 % 2 === 0), the 2nd line will be "thick" (1 % 2 = 1).
Not nice.
ALternatively we can make "thin" lines for "hour=0.5" & "hour=1.5", and "thick" line for "hour=1".
What do you think?
Also, can we turn a "frequency" - that "2" value - into an option like `grid_line_?????` to define which lines should be "thin"?

Reply:
done

Reply:
We still have an issue here.
If hours_to_show = 2, we will get the left picture (thick line on 125, thin on 375), but we need to get the right picture (thick line on 250):
![image](https://github.com/user-attachments/assets/fe059c93-45d9-4cf5-87c3-c1306eb08454)

Reply:
yeah that's the 0.5 in this line const x = xRatio * (i + 0.5);
If I remove it the first line is kinda hidden in the border and I find it wierd, that's why there is an offset

Reply:
I am trying now to adapt the algorithm to a fractional `hours_to_show`. For int values already made it, but for fractional more difficult story.

Reply:
Please check my addition below (currently last comment in the thread).

Reply:
refactored

Reply:
Very good.
Look, I have a little experience in JS - so have lots of doubts.
In C we need to provide things like
```
foo = a / float(b);
```
to avoid undesired conversions.
Is it properly handled in JS? Don't we need to write additional checks or explicit conversions ?

Reply:
The type in js is "number", it can be either int, float, long ... 
Here, js automatically assign the digits (if you want an int for example, you need to explicitly remove the digits)

Reply:
So, no additional conversions are needed in this code? Will it not loose a precision? Critical for the algorithm.

Reply:
I've double-checked
js is not the best when a LOT of digits are involved. Here it is not a big deal as it is only light rounding problems.
So there is no explicit conversions needed.

Reply:
Ok. Let us proceed with testing with a generated js-file. Please participate if possible).

Reply:
I'm currently using my 3 PRs in my daily driving dashboard. Is that what you mean by participate in the testing ?

Reply:
Do you mean PRs after all our joint corrections?

Reply:
yeah I did a merge on a local branch this weekend, built it and pushed it to my HA.
For this one and the labels, It's rocking.
For the loader, I might come back to the code to check if my condition to show to loader is right.

Reply:
Good, at least grid lines and legend labels works fine. Hope will test them too in a few days. These are very important PRs.